### PR TITLE
Add a Deployment rollback step

### PR DESCRIPTION
**Pull request by ahpook**
This step will roll back a Deployment running on a GKE cluster.

### CODE REVIEW

**Comment by ahpook** [comment]
@hunner wdyt? i see you're making new repos rather than additional steps in this one, should i do the same?

File: steps/rollout-undo/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# gcp-step-cloud-run-deploy
```
Comment (hunner) [inline steps/rollout-undo/README.md]: This is still the old readme

File: steps/rollout-undo/spec.schema.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        }
+    },
+    "namespace": {
+        "type":"string",
+        "description":"The namespace in which the deployment is running"
+    },
+    "deployment": {
+        "type":"string",
+        "description":"The name of the deployment to operate on"
+    },
```
Comment (hunner) [inline steps/rollout-undo/spec.schema.json:26]: Also `cluster` and `zone`

File: steps/rollout-undo/spec.schema.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
+
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "properties": {
+    "credentials": {
```
Comment (hunner) [inline steps/rollout-undo/spec.schema.json]: Something like
```json
"google": {
  "type": "object",
  "x-relay-connectionType": "gcp",
  "description": "A Relay Google Cloud Platform (GCP) connection to use",
  "properties": {
    "serviceAccountKey": {
      "type": "string"
    }
  },
  "required": [
    "serviceAccountKey"
  ]
},
```

File: steps/rollout-undo/step.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  - summary: Undo a rollout of the nginx deployment in namespace production
+    content: 
+      apiVersion: v1
+      kind: Step
+      name: gcp-rollback
+      image: relaysh/gcp-step-rollout-undo
+      spec:
+        credentials: !Connection { type: gcp, name: relay-service-account }
+        deployment: !Parameter serviceName
+        namespace: !Parameter namespace
```
Comment (hunner) [inline steps/rollout-undo/step.yaml:36]: And `cluster` and `zone`

**Review by hunner — COMMENTED** [review]
I'm breaking repos out into separate services, and that also means that fewer containers are rebuilt on push. For this one, I'm not sure what the separate repo would be called... gcp-kubectl or something? It's probably fine to leave it here and break stuff out later when we come back to it.

File: steps/rollout-undo/spec.schema.json [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      "description":"The namespace in which the deployment is running"
+  },
+  "deployment": {
+      "type":"string",
+      "description":"The name of the deployment to operate on"
+  },
+  "cluster": {
+      "type":"string",
+      "description":"The name of the cluster where the deployment is running"
+   "zone": {
```
Comment (hunner) [inline steps/rollout-undo/spec.schema.json]: Looks like the syntax is off here

File: steps/rollout-undo/step.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+examples:
+  - summary: Undo a rollout of the nginx deployment in namespace production
+    content: 
+      apiVersion: v1
+      kind: Step
+      name: gcp-rollback
+      image: relaysh/gcp-step-rollout-undo
+      spec:
+        credentials: !Connection { type: gcp, name: relay-service-account }
```
Comment (hunner) [inline steps/rollout-undo/step.yaml]: This can be `google` now

**Review by hunner — APPROVED** [review]
Spotted a couple small things, but 👍

**Comment by ahpook** [comment]
I think this should be good now, thanks @hunner

**Pull request merged**